specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json

```diff
@@ -0,0 +1,489 @@
+          },
+          {
+            "$ref": "./vnfmanager.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./vnfmanager.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Registration Key information for the Vnf Manager device.",
+            "schema": {
+              "$ref": "#/definitions/DeviceRegistrationKey"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./vnfmanager.json#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get VnfManager devices Registration Key": {
+            "$ref": "./examples/DeviceGetRegistrationKey.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "VnfManagerDevice": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VnfManagerDevicePropertiesFormat",
+          "description": "Vnf Manager device Properties."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./vnfmanager.json#/definitions/Resource"
+        }
+      ],
+      "description": "VnfManager device  resource."
+    },
+    "VnfManagerDevicePropertiesFormat": {
+      "description": "Vnf Manager device properties.",
+      "properties": {
+        "resourceGuid": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The resource GUID property of the VnfManagerDevice resource."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./vnfmanager.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the VnfManager device resource."
+        },
+        "dataBoxEdgeDevice": {
+          "readOnly": true,
+          "$ref": "./vnfmanager.json#/definitions/SubResource",
+          "description": "The reference to the databox edge device"
+        },
+        "sku": {
+          "type": "string",
+          "description": "The sku of the Vnf Manager device"
+        },
+        "mecVNFs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VNFRef"
+          },
+          "description": "The list of VNFs deployed on the Vnf Manager device."
+        },
+        "mecRegistrationKey": {
+          "type": "string",
+          "description": "The registration key for the Vnf Manager device resource"
+        }
+      }
+    },
+    "DeviceRegistrationKey": {
+      "description": "Vnf Manager Device registration key",
+      "properties": {
+        "registrationKey": {
```

Comment:
>registrationKey [](start = 9, length = 15)

annotate with x-ms-secret: true

Reply:
is this same as mecRegistrationKey in PUT.. ? probably use same name in both ?

---
In reply to: [367744014](https://github.com/Azure/azure-rest-api-specs/pull/8162#discussion_r367744014) [](ancestors = 367744014)

Reply:
Yes. I have changed and replace the same name.

Reply:
I am getting the Schema Violation when I am using the property "x-ms-secret".
ERROR: Schema violation: Additional properties not allowed: x-ms-secret
    - file:///C:/Users/neeleshk/azure-rest-api-specs/specification/vnfmanager/resource-manager/Microsoft.VnfManager/preview/2020-01-01-preview/device.json:458:8 ($.definitions.DeviceRegistrationKey.properties.mecRegistrationKey)

I have not added this property but fixed other comments.